modules/widgets/top-posts.php

```diff
@@ -488,12 +488,14 @@ function get_by_views( $count, $args ) {
 			$days = 2;
 		}
 
-		$post_view_posts = stats_get_csv( 'postviews', array( 'days' => absint( $days ), 'limit' => 11 ) );
-		if ( ! $post_view_posts ) {
+		$post_view_posts = stats_get_from_restapi( array(), 'top-posts?max=10&summarize=1&num=' . absint( $days ) );
```

Comment:
Yes, that commit message wasn't very helpful.

The widget can indeed be set to return 10 posts maximum. However, among those 10 posts you'll often find the home page, and that home page is not displayed in the Top Posts Widget.

That means that if you do a query for 10 posts, you may end up with only 9 unique posts that can be used in the widget.

To avoid displaying only 9 posts when the site owner asked for 10 posts in the widget settings, we decided to query for one more than the maximum (11). We then make sure to respect the specified `$count` by adding that limit to the array of posts that we build later on.

I hope this clarifies things a bit. I'll add a link to this comment in the original commit, just in case this conversation comes up again in the future :)